kornia/geometry/warp/homography_warper.py

```diff
@@ -129,67 +180,28 @@ def forward(  # type: ignore
             >>> warper.precompute_warp_grid(homography)
             >>> output = warper(input)  # NxCxHxW
         """
-        warped_grid: torch.Tensor
+        _warped_grid = self._warped_grid
         if src_homo_dst is not None:
-            if not src_homo_dst.device == patch_src.device:
-                raise TypeError("Patch and homography must be on the same device. \
-                                Got patch.device: {} src_H_dst.device: {}."
-                                .format(patch_src.device, src_homo_dst.device))
-            warped_grid = self.warp_grid(src_homo_dst)
-        elif self._warped_grid is not None:
-            if not self._warped_grid.device == patch_src.device:
+            warped_patch = homography_warp(
+                patch_src, src_homo_dst, (self.height, self.width), mode=self.mode,
+                padding_mode=self.padding_mode, align_corners=self.align_corners,
+                normalized_coordinates=self.normalized_coordinates)
+        elif _warped_grid is not None:
+            if not _warped_grid.device == patch_src.device:
                 raise TypeError("Patch and warped grid must be on the same device. \
-                                Got patch.device: {} warped_grid.device: {}. Wheter \
-                                recall precompute_warp_grid() with the correct device \
-                                for the homograhy or change the patch device."
-                                .format(patch_src.device, self._warped_grid.device))
-            warped_grid = self._warped_grid
+                                 Got patch.device: {} warped_grid.device: {}. Wheter \
+                                 recall precompute_warp_grid() with the correct device \
+                                 for the homograhy or change the patch device.".format(
+                                patch_src.device, _warped_grid.device))
+            warped_patch = F.grid_sample(
```

Comment:
probably to encapsulate somewhere `grid_sample` we could replace this call to `remap`